object-store-client-common/src/main/scala/uk/gov/hmrc/objectstore/client/model.scala

```diff
@@ -67,3 +67,8 @@ final case class ObjectSummary(
   contentMd5: String,
   lastModified: Instant
 )
+
+object ObjectExpirationPeriod extends Enumeration {
```

Comment:
Can we just create a sealed trait rather than use `Enumeration`? Also separate the value, to keep scala style.

Reply:
Will we get any benefit by defining an ADT for this? 
This is just a plain data going over the wire and we have no business rules in atleast object-store-client.

Reply:
It's the standard we usually follow. Enumeration doesn't catch partial matches, uses reflection etc.

Reply:
yea, I know partial matches doesn't work with enums but are we going to use any partial matches here in the client for the expiration period? I guess not. It is just a plain data value being passed from client to object-store service.

Usually I favour ADTs over enums but if it is just plain data value like in this case then I think ADT is just an overkill for it.  Enums fits the current purpose.

Reply:
Done